app/models/solidus_stripe/payment_method.rb

```diff
@@ -7,37 +7,143 @@ class PaymentMethod < ::Spree::PaymentMethod
 
     validates :available_to_admin, inclusion: { in: [false] }
 
-    def partial_name
-      "stripe"
-    end
+    concerning :Actions do
+      def actions
+        %w[capture void credit]
+      end
 
-    alias cart_partial_name partial_name
-    alias product_page_partial_name partial_name
+      def can_capture?(payment)
+        payment.pending?
+      end
+
+      def can_void?(payment)
+        payment.pending?
+      end
 
-    def payment_source_class
-      PaymentSource
+      def can_credit?(payment)
+        payment.completed? && payment.credit_allowed > 0
+      end
     end
 
-    def gateway_class
-      Gateway
+    concerning :Configuration do
+      def partial_name
+        "stripe"
+      end
+
+      alias cart_partial_name partial_name
+      alias product_page_partial_name partial_name
+      alias risky_partial_name partial_name
+
+      def source_required?
+        false
+      end
+
+      def payment_source_class
+        PaymentSource
+      end
+
+      def gateway_class
+        Gateway
+      end
     end
 
-    def create_profile(payment)
-      payment_intent = payment.source.payment_intent
+    concerning :Order do
+      def find_in_progress_payment_for(order)
+        payments = order.payments
+          .where.not(state: %w[completed invalid void]) # in_progress
+          .where(payment_method: self)
+          .order(:created_at)
+          .entries
+
+        *old_payments, payment = payments
+        old_payments.each(&:invalidate!)
```

Comment:
Oh right!

The underlying issue is that to this date we don't have test coverage for this, but I don't want to commit to strict coverage until profiles are added to the mix, as things can still change considerably with that addition.

Reply:
TBH, I don't feel comfortable with this approach. IMHO, the test suite should be something to change along the way. Otherwise, we're moving on top of shifting sands, and it's too easy to avoid covering some behavior on an ex-post read. At least, I think we should keep a list of things that need to be accounted for.

Reply:
@loicginoux extracted the comment for a later fix in https://github.com/solidusio/solidus_stripe/issues/200 so we don't forget.

The discussion about how to approach development was taken to a series of online meetings and beyond the scope of this comment thread. I'll leave an update at the end of this PR later.